aten/src/THC/generic/THCTensorRandom.cu

```diff
@@ -37,33 +37,17 @@ void THCTensor_(multinomial)(struct THCState *state,
                               int n_sample,
                               int with_replacement)
 {
-  THCAssertSameGPU(THCTensor_(checkGPU)(state, 2, self, prob_dist));
```

Comment:
I'm a little confused: it looks like you added new checks in `multinomial_out_cpu` but removed the ones here. Do the error checks still exist in CUDA?

Also, could you add a test that tests that the error checking is consistent in both CPU/CUDA?

Reply:
Fixed